packages/block-editor/src/components/block-edit/use-attribute-picker.js

```diff
@@ -0,0 +1,138 @@
+ * It can be used to write hooks like `useColors`.
+ *
+ * @see [useColors](/packages/block-editor/README.md#useColors)
+ *
+ * @param {Object}                 attributePickerOptions                           The options object.
+ *
+ * @param {string[]}               attributePickerOptions.names                     The list of attributes to manage.
+ *
+ * @param {*[]}                    attributePickerOptions.valuesEnum                The list of values attributes can be set to.
+ *                                                                                   Setting an attribute to a custom value not on
+ *                                                                                   this list will set the attribute's value to
+ *                                                                                   `undefined` and the corresponding `custom${ upperFirst( name ) }`
+ *                                                                                   attribute to the custom value.
+ *
+ * @param {Function}               [attributePickerOptions.findEnumValue]           Function for comparing an enum value to the
+ *                                                                                   value passed to the setter, when looking for
+ *                                                                                   the value in the enum. Defaults to shallow comparison.
+ *
+ * @param {Function}               [attributePickerOptions.mapEnumValue]            Function for mapping the enum value found with
+ *                                                                                   `findEnumValue` before saving it, if found.
+ *                                                                                   Otherwise, the value is assumed to be custom and saved as is.
+ *
+ * @param {*[]}                    [attributePickerOptions.findAndMapEnumValueDeps] List of items that, if they fail a shallow equality test, will
+ *                                                                                   cause new setters to be returned. Useful if your `findEnumValue` or `mapEnumValue` have closures.
+ *
+ * @param {Function}               [attributePickerOptions.mapAttribute]            Function for mapping attributes before returning them, called
+ *                                                                                   with the attribute's or custom attribute's value and the name.
+ *
+ * @param {*[]}                    [attributePickerOptions.mapAttributeDeps]        List of items that, if they fail a shallow equality test, will
+ *                                                                                   cause attributes to be mapped again. Useful if your `mapAttribute` has closures.
+ *
+ * @param {{ string: Function }}   [attributePickerOptions.utilsToBind]             Object with functions to bind to `valuesEnum` and merge into the
```

Comment:
The relationship between utilsToBind and the other properties is not obvious, also the fact that we always bind valuesEnum seems really restrict, I may want to find to the selected value for example.
It's not hard for someone using this hook to create util functions and bind them to properties of the returned object (useMemo call where the returned object is a dependency, and return a new object that uses the returned object and adds the functions with binds to properties returned from useAttributePicker). Maybe we can remove utilsToBind to simplify this component.

Reply:
I thought binding to `valuesEnum` could be common after seeing the current colors implementation. But I wasn't fully convinced by this API either. It feels like it's "forcing" the generalization.